python/cudf/cudf/core/index.py

```diff
@@ -443,6 +443,12 @@ def __init__(self, start, stop=None, name=None):
         self._cached_values = None
 
     def __contains__(self, item):
+        if not isinstance(
+            item, tuple(np.sctypes["int"] + np.sctypes["float"] + [int, float])
```

Comment:
Shouldn't floats return false?

Reply:
I guess we need to check the situation where we're passed something like `3.0`. I think we need to special case floats here and check if they're an integer, i.e. `(3.0).is_integer()` and return False if they're not an integer.

Reply:
Correct, pandas passes the case you mentioned. I think I handle it by modding to `1` though, no?

Reply:
That's an `elif` so it won't run unless I'm missing something

Reply:
oh, oops. Fixing